### ISSUE TITLE
Reference-style links

### ISSUE THREAD

**Issue by invliD** [laptobbe/TSMarkdownParser#56]
The parser currently does not support [reference-style links](https://daringfireball.net/projects/markdown/syntax#link). Is there any plan to support that syntax?

**Comment by Coeur** [comment]
No, I do not have plan to support it in the near future, as it is not doable with regex. A grammar parser/lexer would be needed, so basically a totally different implementation.

**Comment by Coeur** [comment]
It could be done with regex, but with multiple passes, which is not desired most of the time.

> This is [an example] [id] reference-style link.
> 
> [id]: http://example.com/  "Optional Title Here"

**Comment by Coeur** [comment]
Bah... I did multiple passes already for CodeEscaping... so I guess I can do it for reference-style links.

Let me try to work on it for version 3.0.0.

**Comment by invliD** [comment]
@Coeur What's your status with this feature? Need any help?

**Comment by Coeur** [comment]
Will release it this weekend.

**Comment by invliD** [comment]
@Coeur Any news?

**Comment by Coeur** [comment]
I'm so sorry. As you noticed, I didn't work for a few months.

**Comment by invliD** [comment]
@Coeur Did you make some progress? Could you publish your current version so I can finish it if you don't have the time?
